components/style/gecko/values.rs

```diff
@@ -422,6 +423,27 @@ impl GeckoStyleCoordConvertible for ScrollSnapPoint<LengthOrPercentage> {
     }
 }
 
+impl<L> GeckoStyleCoordConvertible for Perspective<L>
+where
+    L: GeckoStyleCoordConvertible,
+{
+    fn to_gecko_style_coord<T: CoordDataMut>(&self, coord: &mut T) {
+        match *self {
+            Perspective::None => coord.set_value(CoordDataValue::None),
+            Perspective::Length(ref l) => l.to_gecko_style_coord(coord),
+        };
+    }
+
+    fn from_gecko_style_coord<T: CoordData>(coord: &T) -> Option<Self> {
+        use gecko_bindings::structs::root::nsStyleUnit;
+
+        if coord.unit() == nsStyleUnit::eStyleUnit_None {
+            return Some(Perspective::None);
+        }
+        Some(Perspective::Length(L::from_gecko_style_coord(coord).unwrap()))
```

Comment:
Shouldn't this use `?` / `map` instead of unwrapping? This is supposedly a fallible operation.

Reply:
There was an `expect` before, I'm not sure this is supposed to ever be called with a value with the wrong unit. But then I don't even understand why this trait can return `None`.

Reply:
Still think it should use `?`...